### PR TITLE
Resource index migration to ArkLib-Android

### PR DESCRIPTION
**Pull request by ShubertMunthali**

### CODE REVIEW

File: app/src/androidTest/java/space/taran/arknavigator/ResourceMetaTest.kt [hunk 1]
```diff
@@ -44,6 +44,6 @@ class ResourceMetaTest {
         } catch (e: Exception) {
             throw RuntimeException(e)
         }
-        assertEquals(computeId(size, file.toPath()), 0xe4c0951c)
+        assertEquals(computeId(size, file.toPath()).crc32, 0xe4c0951c)
     }
 }
```
Comment (mdrlzy) [inline app/src/androidTest/java/space/taran/arknavigator/ResourceMetaTest.kt:49]: It looks like we don't need this test now, it's already in arklib-android

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/scores/PlainScoreStorage.kt [hunk 2]
```diff
@@ -103,8 +104,10 @@ class PlainScoreStorage(
             verifyVersion(version)
             val result = lines.map {
                 val parts = it.split(KEY_VALUE_SEPARATOR)
-                val id = parts[0].toLong()
-                val score = parts[1].toInt()
+                val id = ResourceId.create(
+                    parts[0].toLong(), parts[1].toLong()
+                )
+                val score = parts[2].toInt()
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/scores/PlainScoreStorage.kt:110]: `create` method is needed only for JNI, you can create an object in the usual way.

We need to decide how to store `ResourceId` in text and encapsulate serialization logic inside `ResourceId`

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/scores/PlainScoreStorage.kt [hunk 3]
```diff
@@ -129,7 +132,9 @@ class PlainScoreStorage(
 
         lines.addAll(
             entries.map { (id, score) ->
-                "$id$KEY_VALUE_SEPARATOR$score"
+                val converters = Converters()
+                converters.resourceIdToString(id) +
+                    "$KEY_VALUE_SEPARATOR$score"
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/scores/PlainScoreStorage.kt:137]: We have to encapsulate serialization logic inside `ResourceId`

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/tags/PlainTagsStorage.kt [hunk 4]
```diff
@@ -233,8 +234,10 @@ class PlainTagsStorage(
             val result = lines
                 .map {
                     val parts = it.split(KEY_VALUE_SEPARATOR)
-                    val id = parts[0].toLong()
-                    val tags = tagsFromString(parts[1])
+                    val id = ResourceId.create(
+                        parts[0].toLong(), parts[1].toLong()
+                    )
+                    val tags = tagsFromString(parts[2])
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/tags/PlainTagsStorage.kt:240]: Same as in `ScoreStorage`

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/tags/PlainTagsStorage.kt [hunk 5]
```diff
@@ -262,7 +265,9 @@ class PlainTagsStorage(
             val entries = tagsById.filterValues { it.isNotEmpty() }
             lines.addAll(
                 entries.map { (id, tags) ->
-                    "$id$KEY_VALUE_SEPARATOR ${stringFromTags(tags)}"
+                    val converters = Converters()
+                    converters.resourceIdToString(id) +
+                        "$KEY_VALUE_SEPARATOR ${stringFromTags(tags)}"
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/tags/PlainTagsStorage.kt:270]: Same as in `ScoreStorage`

File: app/src/test/java/space/taran/arknavigator/stub/TestData.kt [hunk 6]
```diff
@@ -51,10 +52,10 @@ object TestData {
...
 }
 
-const val R1 = 1L
-const val R2 = 2L
-const val R3 = 3L
-const val R4 = 4L
+val R1 = ResourceId.create(1L, 1L)
+val R2 = ResourceId.create(2L, 2L)
+val R3 = ResourceId.create(3L, 3L)
+val R4 = ResourceId.create(4L, 4L)
```
Comment (mdrlzy) [inline app/src/test/java/space/taran/arknavigator/stub/TestData.kt:58]: You can create objects in usual way

**Comment by ShubertMunthali** [comment]
> @ShubertMunthali It's better to do task in branch based on main branch
> 
> This PR also includes changes for forgetting roots, which hinders review

Alright

**Pull request closed without merging**